src/Domain/CoordinateMaps/TimeDependent/Shape.hpp

```diff
@@ -251,7 +251,7 @@ class Shape {
       gsl::not_null<std::array<DataVector, 3>*> target_coords,
       gsl::not_null<std::array<DataVector, 3>*> frame_vel,
       gsl::not_null<tnsr::Ij<DataVector, 3, Frame::NoFrame>*> jac,
-      const std::array<DataVector, 3> source_coords, double time,
+      const std::array<DataVector, 3> source_coords, const double time,
```

Comment:
Make `source_coords` a reference (you'll have to edit the metafunction as well). Also no `const` for `time` in the declaration

Reply:
Yes, I was unsure about this. The coords_frame_velocity_jac function currently uses this function with the reference return coords object and this argument having the same address (the source coords are overwritten with the target coords). This  is why I copied it to be safe. It should be fine either way as the arg is used only before the reference return object is touched by it still feels uncomfortable to have it this way. What is the usual C++ policy on this?

Reply:
Ohh I see now. In `src/Domain/CoordinateMaps/CoordinateMapHelpers.hpp` for the `apply_map` function, we just pass the coords through the `gsl::not_null` part and use that as the source coords. We can probably do the same thing here